packages/firestore/src/core/sync_engine.ts

```diff
@@ -239,6 +240,19 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     });
   }
 
+  private synchronizeLocalView(queryData: QueryData): Promise<void> {
+    const query = queryData.query;
+
+    return this.localStore.executeQuery(query).then(docs => {
+      return this.localStore
+        .remoteDocumentKeys(queryData.targetId)
+        .then(async remoteKeys => {
+          const queryView = this.queryViewsByTarget[queryData.targetId];
```

Comment:
As alluded to elsewhere, I think queryView could be undefined for targets from other tabs...  Depending on how you fix that, it may make sense to add an assert here (the following line will probably crash anyway, but an assert will be clearer).